go/fn/resourcelist.go

```diff
@@ -57,6 +57,21 @@ type ResourceList struct {
 	Results Results `yaml:"results,omitempty" json:"results,omitempty"`
 }
 
+// CheckResourceDuplication checks the GVKNN of resourceList.items to make sure they are unique. It returns errors if
+// found more than one resource having the same GVKNN.
+func CheckResourceDuplication(rl *ResourceList) error {
```

Comment:
Should we check this by default somewhere?

Reply:
Yes, I plan to add this duplication check in the `Run(p ResourceListProcessor...)` and `Execute(p ResourceListProcessor...)`. This will require some additional work on error handling and code refactoring, because the returned error of `CheckResourceDuplication` is out of the function scope so it won't be shown in `ResourceList.Results` https://github.com/GoogleContainerTools/kpt-functions-sdk/blob/master/go/fn/run.go#L39

Reply:
Currently, this function is called by functions directly. 
https://github.com/GoogleContainerTools/kpt-functions-catalog/pull/844/files#diff-2b0b1f52afe9e1c4943996b7c2f696f3b641bfc6e0e5beef48d273e72b2b9d19R43